### PR TITLE
[RFR] Bring some fixes for autocomplete fields

### PR DESCRIPTION
**Pull request by jpetitcolas**

### CODE REVIEW

**Comment by fzaninotto** [comment]
Since your code isn't very easy to follow, a couple unit tests would help.

**Comment by jpetitcolas** [comment]
This PR simply reverts some changes made in #16.

Final aim is to allow a function as `filters`. Adding a unit test on it.

File: lib/Queries/ReadQueries.js [hunk 1]
```diff
@@ -197,7 +192,7 @@ class ReadQueries extends Queries {
                 1,
                 reference.perPage(),
                 typeof(filters) === 'function' ? filters(search) : filters,
-                null,
+                reference,
```
Comment (jeromemacias) [inline lib/Queries/ReadQueries.js]: `getRawValues` method take an object wich is a list of field indexed by name.

IMO, this will be ignore.

Comment (jpetitcolas) [inline lib/Queries/ReadQueries.js]: You are true. Fixed, even if we always passed an empty object before. I passed an argument for better consistency.

**Comment by jpetitcolas** [comment]
Unit test added.

File: tests/lib/Queries/ReadQueriesTest.js [hunk 2]
```diff
@@ -237,4 +237,29 @@ describe('ReadQueries', () => {
...
         });
     });
+
+    describe('getAllReferencedData', () => {
+        var getRawValuesMock;
+        beforeEach(function() {
+             getRawValuesMock = sinon.mock(readQueries);
+        });
+
+        it.only('should execute filters function with current search parameter if filters is a function', () => {
```
Comment (jeromemacias) [inline tests/lib/Queries/ReadQueriesTest.js]: Can you remove the `.only` ?

**Pull request merged**